src/NServiceBus.Core/Pipeline/ConnectorContextExtensions.cs

```diff
@@ -61,6 +61,14 @@ public static IIncomingPhysicalMessageContext CreateIncomingPhysicalMessageConte
         }
 
         /// <summary>
+        /// Creates a <see cref="IIncomingPhysicalMessageContext"/> based on the current context.
+        /// </summary>
+        public static IIncomingPhysicalMessageContext CreateIncomingPhysicalMessageContext(this StageConnector<ITransportReceiveContext, IIncomingPhysicalMessageContext> stageConnector, IncomingMessage incomingMessage, ITransportReceiveContext sourceContext)
```

Comment:
> I don't think we should allow for it.

@SzymonPobiega can u explain why u think this specific part of the pipeline should not be replaceable when compared to all the other parts?